source/loading.c

```diff
@@ -32,58 +32,20 @@
 
 #include <png.h>
 
-void delete_entry(Entry_s * entry, bool is_file)
+void copy_texture_data(C3D_Tex* texture, const u16* src, const Entry_Icon_s* current_icon)
 {
-    if(is_file)
-        FSUSER_DeleteFile(ArchiveSD, fsMakePath(PATH_UTF16, entry->path));
-    else
-        FSUSER_DeleteDirectoryRecursively(ArchiveSD, fsMakePath(PATH_UTF16, entry->path));
-}
-
-u32 load_data(char * filename, Entry_s entry, char ** buf)
-{
-    if(entry.is_zip)
-    {
-        return zip_file_to_buf(filename+1, entry.path, buf); //the first character will always be '/' because of the other case
-    }
-    else
-    {
-        u16 path[0x106] = {0};
-        strucat(path, entry.path);
-        struacat(path, filename);
-
-        return file_to_buf(fsMakePath(PATH_UTF16, path), ArchiveSD, buf);
-    }
-}
-
-// Function taken and adapted from https://github.com/BernardoGiordano/Checkpoint/blob/master/3ds/source/title.cpp
-C2D_Image * loadTextureIcon(Icon_s *icon)
-{
-    if(icon == NULL)
-        return NULL;
-
-    C2D_Image * image = calloc(1, sizeof(C2D_Image));
-    C3D_Tex* tex = malloc(sizeof(C3D_Tex));
-    static const Tex3DS_SubTexture subt3x = { 48, 48, 0.0f, 48/64.0f, 48/64.0f, 0.0f };
-    image->tex = tex;
-    image->subtex = &subt3x;
-    C3D_TexInit(image->tex, 64, 64, GPU_RGB565);
-
-    u16* dest = (u16*)image->tex->data + (64-48)*64;
-    u16* src = icon->big_icon;
+    u16* dest = (u16*)texture->data + current_icon->y * texture->width + current_icon->x * 8;
```

Comment:
readability: add parens for explicit order of operations or break this up into multiple lines